nix/ocaml.nix

```diff
@@ -85,16 +91,150 @@ let
     [ zlib bzip2 gmp openssl libffi ]
     ++ lib.optional (!(stdenv.isDarwin && stdenv.isAarch64)) jemalloc;
 
+  dune-nix = inputs.dune-nix.lib.${pkgs.system};
+
+  base-libs = dune-nix.squashOpamNixDeps scope.ocaml.version
+    (pkgs.lib.attrVals (builtins.attrNames implicit-deps) scope);
+
+  dune-description = pkgs.stdenv.mkDerivation {
+    pname = "dune-description";
+    version = "dev";
+    src = pkgs.lib.sources.sourceFilesBySuffices ../src [
+      "dune"
+      "dune-project"
+      ".inc"
+      ".opam"
+    ];
+    phases = [ "unpackPhase" "buildPhase" ];
+    buildPhase = ''
+      files=$(ls)
+      mkdir src
+      mv $files src
+      cp ${../dune} dune
+      cp ${../dune-project} dune-project
+      ${
+        inputs.describe-dune.defaultPackage.${pkgs.system}
+      }/bin/describe-dune > $out
+    '';
+  };
+
+  duneDescLoaded = builtins.fromJSON (builtins.readFile dune-description);
+  info = dune-nix.info duneDescLoaded;
+  allDeps = dune-nix.allDeps info;
+  commonOverrides = {
+    DUNE_PROFILE = "dev";
+    buildInputs = [ base-libs ] ++ external-libs;
+    nativeBuildInputs = [ ];
+  };
+  packageHasSrcApp =
+    dune-nix.packageHasUnit ({ src, ... }: pkgs.lib.hasPrefix "src/app/" src);
+  sqlSchemaFiles = with inputs.nix-filter.lib;
+    filter {
+      root = ../src/app/archive;
+      include = [ (matchExt "sql") ];
+    };
+
+  granularBase =
+    dune-nix.outputs' commonOverrides ./.. allDeps info packageHasSrcApp;
+  vmOverlays = let
+    commit = inputs.self.sourceInfo.rev or "<dirty>";
+    commitShort = builtins.substring 0 8 commit;
+    cmdLineTest = ''
+      mina --version
+      mv _build/default/src/test/command_line_tests/command_line_tests.exe tests.exe
+      chmod +x tests.exe
+      export TMPDIR=tmp # to align with janestreet core library
+      mkdir -p $TMPDIR
+      export MINA_LIBP2P_PASS="naughty blue worm"
+      export MINA_PRIVKEY_PASS="naughty blue worm"
+      export MINA_KEYS_PATH=genesis_ledgers
+      mkdir -p $MINA_KEYS_PATH
+      echo '{"ledger":{"accounts":[]}}' > $MINA_KEYS_PATH/config_${commitShort}.json
+      ./tests.exe --mina-path mina
+    '';
+  in [
+    (dune-nix.testWithVm { } "mina_net2" [ pkgs.libp2p_helper ])
```

Comment:
Does this `testWithVm` somehow use the `vmOverlays` above? It seems that `vmOverlays` is only passed to `granularOverlay`.